### PR TITLE
institute tokstr

### PR DESCRIPTION
**Pull request by vixie**
as discussed.

### CODE REVIEW

File: tokstr.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		printf("\t\"%s\"\n", t);
+	tokstr_last(&ts);
+
+ * will output "this", "is", and "test".
+ */
+
+struct tokstr;
+typedef struct tokstr *tokstr_t;
+
+tokstr_t tokstr_buffer(const char *source, size_t size);
```
Comment (djw1149) [inline tokstr.h:20]: Needs some function comments on the variations between _buffer and _string

**Comment by djw1149** [comment]
I agree with your "second thoughts" that the read_config() change should be reverted.

Again, looking at the overall changes adopting tokstr, I don't think they are worth it.  It's a small win with added complexity overall.

**Review by djw1149 — CHANGES_REQUESTED** [review]
I agree with your "second thoughts" that the read_config() change should be reverted.

Again, looking at the overall changes adopting tokstr, I don't think they are worth it. It's a small win with added complexity overall.

File: tokstr.h (hunk 1 above)
Comment (vixie) [reply tokstr.h:22]: see 7768f14.

**Review by djw1149 — APPROVED** [review]

**Pull request merged**